runtime/protocol/http-client-engines/test-suite/jvm/test/aws/smithy/kotlin/runtime/http/test/ProxyTest.kt

```diff
@@ -0,0 +1,128 @@
+        // we would expect a customer to configure proxy support on the underlying engine
+        skipEngines = setOf("KtorEngine")
+    ) {
+        engineConfig {
+            val proxyPort = mitmProxy.getMappedPort(8080)
+            proxySelector = ProxySelector { _ ->
+                ProxyConfig.Http("http://127.0.0.1:$proxyPort")
+            }
+        }
+
+        test { _, client ->
+            val req = HttpRequest {
+                // Use http for two reasons (1) mitmproxy wants to sniff traffic by default
+                // and (2) we know the response code will be a redirect if our setup fails.
+                // NOTE: you can still use mitmproxy to proxy https traffic without intercepting it
+                // like any normal proxy by setting the `--ignore-hosts` option to forward all traffic
+                // but that would mean we make an actual request to the origin
+                url(Url.parse("http://aws.amazon.com"))
+                url.path = "/"
+                header("Host", "aws.amazon.com")
+            }
+
+            val call = client.call(req)
+            try {
+                // will be a 301 for http -> https if proxy isn't setup or not configured properly
+                assertEquals(HttpStatusCode.OK, call.response.status, "${client.engine}")
+                val body = call.response.body.readAll()!!.decodeToString()
+                assertEquals("hello proxy", body)
+            } finally {
+                call.complete()
+            }
+        }
+    }
+}
+
+@Testcontainers(disabledWithoutDocker = true)
+@EnabledIfSystemProperty(named = "aws.test.http.enableProxyTests", matches = "true")
+class ProxyAuthTest : AbstractEngineTest() {
+
+    @Container
+    val mitmProxy = mitmProxyContainer("--proxyauth testuser:testpass --set fakeupstream=aws.amazon.com")
+
+    @Test
+    fun testHttpProxyAuth() = testEngines(
+        // we would expect a customer to configure proxy support on the underlying engine
+        skipEngines = setOf("KtorEngine")
+    ) {
+        engineConfig {
+            val proxyPort = mitmProxy.getMappedPort(8080)
+            proxySelector = ProxySelector { _ ->
+                ProxyConfig.Http("http://testuser:testpass@127.0.0.1:$proxyPort")
+            }
+        }
+
+        test { _, client ->
+            val req = HttpRequest {
+                url(Url.parse("http://aws.amazon.com"))
+                url.path = "/"
+                header("Host", "aws.amazon.com")
+            }
+
+            val call = client.call(req)
+            try {
+                // will be a 301 for http -> https if proxy isn't setup or not configured properly
+                assertEquals(HttpStatusCode.OK, call.response.status, "${client.engine}")
+                val body = call.response.body.readAll()!!.decodeToString()
+                assertEquals("hello proxy", body)
+            } finally {
+                call.complete()
+            }
+        }
```

Comment:
**Question**: I see a lot of duplication between this test and the one above it. Can they be refactored to use more common code?